packages/soundxyz/src/Soundxyz.ts

```diff
@@ -163,6 +163,38 @@ export const getProjectFees = async (
   return totalPriceAndFees.total
 }
 
+export const getFees = async (
+  mint: MintActionParams,
+): Promise<{ actionFee: bigint; projectFee: bigint }> => {
+  const { chainId, contractAddress } = mint
+
+  const client = createPublicClient({
+    chain: chainIdToViemChain(chainId),
+    transport: http(),
+  })
+
+  const mintInfoList = (await client.readContract({
+    address: SUPERMINTER_V2,
+    abi: MINT_INFO_LIST_ABI,
+    functionName: 'mintInfoList',
+    args: [contractAddress],
+  })) as MintInfoList
+
+  const mintInfo = mintInfoList[0]
+
+  const totalPriceAndFees = (await client.readContract({
+    address: SUPERMINTER_V2,
+    abi: TOTAL_PRICE_AND_FEES_ABI,
+    functionName: 'totalPriceAndFees',
+    args: [contractAddress, mintInfo.tier, mintInfo.scheduleNum, 1, false], // assume quantity is 1 and hasValidAffiliate is false
+  })) as TotalPriceAndFees
+
+  return {
+    actionFee: totalPriceAndFees.subTotal,
```

Comment:
Should this return `totalPricesAndFees.total`?

Reply:
/*
  https://github.com/soundxyz/sound-protocol/blob/f91f87739c1626b8d2d4e9f02ba573671c627698/contracts/modules/interfaces/ISuperMinterV2.sol#L111C1-L126C6
  struct TotalPriceAndFees {
    uint256 total: (`subTotal + platformTxFlatFee + artistReward + affiliateReward + platformReward`).
    uint256 subTotal: The total price before any additive fees.
    uint256 unitPrice: The price per token.
    uint256 finalArtistFee: The final artist fee (inclusive of `finalArtistReward`).
    uint256 finalAffiliateFee: The total affiliate fee (inclusive of `finalAffiliateReward`).
    uint256 finalPlatformFee: The final platform fee (inclusive of `finalPlatformReward`, `perTxFlat`, sum of `perMintBPS`).
  }
*/

checking the docs, subTotal is basically unitPrice * number of units... I think?